workflows/files/ZWEWRF06.xml

```diff
@@ -0,0 +1,251 @@
+//LOADLIB1 DD DISP=(NEW,CATLG,DELETE),
+//    DSN=${instance-xmem_prefix}.LOADLIB,
+//    DSNTYPE=LIBRARY,SPACE=(CYL,(10,5,5)),
+//    DCB=(RECFM=U),
+//    ${instance-xmem_dasd}
+//PARMLIB1 DD DISP=(NEW,CATLG,DELETE),
+//    DSN=${instance-xmem_prefix}.PARMLIB,
+//    DCB=(LRECL=80,RECFM=FB),SPACE=(CYL,(5,3,5)),
+//    ${instance-xmem_dasd}</inlineTemplate>
+            <submitAs maxRc="0">JCL</submitAs>
+            <maxLrecl>1024</maxLrecl>
+            <!-- zosmfOutput -->
+        </template>
+    </step>
+    <step name="copy_artifacts" optional="false">
+        <title>Copy artifacts</title>
+        <description>Copies the necessary artifacts (load modules, parmlib members, etc). Also copies the STC to the proclib.</description>
+        <prereqStep name="allocate_xmem"/>
+        <!--variableValues-->
+        <!-- condition -->
+        <instructions substitution="false">Run this step to populate the previously allocated data sets and copy the STC procedures.</instructions>
+        <weight>1</weight>
+        <skills>z/OS System Programmer</skills>
+        <autoEnable>true</autoEnable>
+        <canMarkAsFailed>false</canMarkAsFailed>
+        <template>
+            <inlineTemplate substitution="true">set -x
+
+cd "${instance-zowe_install_path}/scripts/utils"
+sh zowe-install-xmem.sh "${instance-zowe_dsn_prefix}" "${instance-xmem_prefix}.LOADLIB" "${instance-xmem_prefix}.PARMLIB" "${instance-proclib}"
+exit $?</inlineTemplate>
+            <submitAs maxRc="0">shell-JCL</submitAs>
+            <maxLrecl>1024</maxLrecl>
+            <!-- zosmfOutput -->
+        </template>
+    </step>
+    <step name="apf_auth" optional="false">
+        <title>APF Authorize Load Library</title>
+        <description>This step will dynamically APF authorize the XMEM LOADLIB</description>
+        <prereqStep name="copy_artifacts"/>
+        <!--variableValues-->
+        <!-- condition -->
+        <instructions substitution="true">Run this step to dynamically APF authorize the load library.
+If you want the APF, to be permanent, add following library to your system authorization: ${instance-xmem_prefix}.LOADLIB</instructions>
+        <weight>1</weight>
+        <skills>z/OS System Programmer</skills>
+        <autoEnable>true</autoEnable>
+        <canMarkAsFailed>false</canMarkAsFailed>
+        <rest>
+            <httpMethod>PUT</httpMethod>
+            <uriPath substitution="true">/zosmf/restconsoles/consoles/${_step-stepOwnerUpper}</uriPath>
+            <requestBody substitution="true">#if(${instance-xmem_dasd.startsWith("STORCLAS")})
+#set($dasd = "SMS")
+#elseif(${instance-xmem_dasd.startsWith("VOL=SER=")})
+#set($dasd = "VOLUME=${instance-xmem_dasd.substring(8)}")
+#end
+{"cmd":"SETPROG APF,ADD,DSNAME=${instance-xmem_prefix}.LOADLIB,${dasd}"}</requestBody>
+            <expectedStatusCode>200</expectedStatusCode>
+            <!-- REST mappings -->
+        </rest>
+    </step>
+    <step name="modify_scheduler" optional="false">
```

Comment:
see my comments about updating an active PARMLIB. We must provide an alternative path as this one is very dangerous..

Reply:
I will remove this option completely and just add the instructions to the step how to do it permanently